custom_model_runner/datarobot_drum/drum/language_predictors/r_predictor/score.R

```diff
@@ -210,19 +211,20 @@ outer_predict <- function(target_type, input_filename=NULL, binary_data=NULL, mo
         }
     }
 
-    if (!is.null(input_filename)) {
-        if (!isFALSE(read_input_data_hook)) {
-            data <- read_input_data_hook(input_filename)
-        } else if (grepl("\\.mtx$", input_filename)) {
-            data <- as.data.frame(as.matrix(readMM(input_filename)))
-        } else {
-            tmp = readChar(input_filename, file.info(input_filename)$size)
-            data <- read.csv(text=gsub("\r","", tmp, fixed=TRUE))
-        }
+    if (!isFALSE(read_input_data_hook)) {
+        data <- read_input_data_hook(binary_data)
+    } else if (!is.null(mimetype) && mimetype == "text/mtx") {
+        tmp_file_name <- tempfile()
+        f <- file(tmp_file_name, "wb")
```

Comment:
Isn't it possible in R to read the binary data directly from memory? Looks like a redundant step of writing to the disk.

Reply:
Unfortunately I'm not able to find smth like `BytesIO` in Python.

`readMM` can read from the R `connections` which all behave like a file: https://stat.ethz.ch/R-manual/R-devel/library/base/html/connections.html